code/streaming/Hacl.Streaming.Blake2.fst

```diff
@@ -131,29 +141,26 @@ let stateful_blake2 (a : alg) (m : m_spec) : I.stateful unit =
 /// ============
 
 inline_for_extraction noextract
-let key_size_ty (a : alg) = key_size:nat{0 <= key_size /\ key_size <= Spec.max_key a}
+let key_size (a : alg) = kk:nat{0 <= kk /\ kk <= Spec.max_key a}
 
 inline_for_extraction noextract
-let key_size_t (a : alg) (is_zero: bool) =
-  key_size:U32.t{
-    0 <= U32.v key_size /\ U32.v key_size <= Spec.max_key a /\
-    (is_zero ==> U32.v key_size = 0)}
+let key_size_t (a : alg) =
+  key_size:U32.t{0 <= U32.v key_size /\ U32.v key_size <= Spec.max_key a}
 
 /// Defining stateful keys
 inline_for_extraction noextract
-let stateful_key_t (a : alg) (no_key : bool)
-                   (key_size : key_size_t a no_key) : Type =
-  if no_key then unit else b:B.buffer uint8 { B.length b == U32.v key_size}
+let stateful_key_t (a : alg) (key_size : key_size a) : Type =
+  if key_size = 0 then unit else b:B.buffer uint8 { B.length b == key_size}
```

Comment:
iirc, the latest conclusion was to leave this to a further PR, in order to avoid duplication in the exposed functions (and to actually expose keyed blake in the generated code)? I might be misrembering

Reply:
no problem with leaving this to a further PR